modules/installation-guide/partials/proc_configuring-workspaces-tolerations.adoc

```diff
@@ -0,0 +1,13 @@
+
+[id="configuring-workspaces-tolerations"]
+= Configuring workspaces tolerations
+
+This section describes how to configure link:https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/[`taint tolerations`] for Pods of {prod-short} workspaces.
+
+.Procedure
+
+{prod-short} uses the `CHE_WORKSPACE_POD_TOLERATIONS__JSON` environment variable to configure `tolerations`. This variable can contain a JSON Array string representing an array of taint tolerations, or `NULL` to disable it. The JSON Array objects has to follow this link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#toleration-v1-core[spec].
```

Comment:
It is really Procedure?

It seems to be a detailed description for `CHE_WORKSPACE_POD_TOLERATIONS__JSON`, the procedure is described in 
https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/#understanding-che-server-advanced-configuration-using-the-operator_che

So, it may be refactored or at least add a reference: to configure this env var following instructions from https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/#understanding-che-server-advanced-configuration-using-the-operator_che

Reply:
To be honest, I am a little out of my comfort zone here and I pretty much [copied what was done](https://www.eclipse.org/che/docs/che-7/installation-guide/configuring-workspaces-nodeselector/) for `nodeSelector`.  That having been said, I agree with your comment.  I am willing to do this right, but I might need some coaching.

So, should I simply create a new entry under the Che Server [env table](https://www.eclipse.org/che/docs/che-7/installation-guide/advanced-configuration-options-for-the-che-server-component/#LChe-server) and delete the procedure page?

Reply:
The new entry is supposed to appear there automatically, since it's generated from https://github.com/eclipse/che/blob/master/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties

@themr0c Could you confirm? Is one who modified che.properties supposed to regenerate that table?

Reply:
@themr0c @sleshchenko I think we are talking about this table: https://github.com/eclipse/che-docs/blob/master/modules/installation-guide/examples/system-variables.adoc.  Can I go ahead and modify it?

Reply:
I think it's supposed to be generated by `.tools/environment_docs_gen.sh`
```
~/projects/che-docs/ dashboardNext *1 !1 ?2 > .tools/environment_docs_gen.sh                                                  13:55:06
Trying to read current product version from /home/sleshche/projects/che-docs/antora-playbook.yml...
Detected version: 7.23.x
Fetching property files content from GitHub...
Fetching content done. Trying to parse it.
   Found begin of topic: Che server
   Found begin of topic: Authentication parameters
   Found begin of topic: Internal
   Found begin of topic: Kubernetes Infra parameters
   Found begin of topic: OpenShift Infra parameters
   Found begin of topic: Experimental properties
   Found begin of topic: Configuration of major "/websocket" endpoint
   Found begin of topic: CORS settings
   Found begin of topic: Factory defaults
   Found begin of topic: Devfile defaults
   Found begin of topic: Che system
   Found begin of topic: Workspace limits
   Found begin of topic: Users workspace limits
   Found begin of topic: Organizations workspace limits
   Found begin of topic: Organizations notifications settings
   Found begin of topic: Multi-user-specific OpenShift infrastructure configuration
   Found begin of topic: Keycloak configuration
   Found begin of topic: RedHat Che specific configuration
Processing done. Output file is /home/sleshche/projects/che-docs/modules/installation-guide/examples/system-variables.adoc
```
So, go ahead and consider improving https://github.com/eclipse/che/pull/18691/files#diff-deca31b9cca47d5b0356dd75fc0fd0c8cb980f3aa029da27b475bdbb37353eb0R689

After that I don't know if you need to provide a regenerated table or it will be done automatically once PR is merged.

Reply:
Yes, `.tools/environment_docs_gen.sh` generates the  `modules/installation-guide/examples/system-variables.adoc` file.

On `7.24.x` and `master` branch, `.tools/environment_docs_gen.sh` detects the version from the `prod-ver` AsciiDoc attriute in `antora-playbook.yaml`.

As of today, "Detected version:" should be `7.24.x` for a PR to `7.24.x` branch, `7.25.x` for a PR to `master` branch. If not, then please rebase on the target branch.

Note that changes on this file cannot be cherry-picked into two branches with different `prod-ver`. We'll need one PR for each target version.

(Previously, the version was detected from the VERSION file which was erroneously maintained. Let's forget about this if the target version is Che 7.24.x)